src/client/Microsoft.Identity.Client/ApiConfig/Executors/ConfidentialClientExecutor.cs

```diff
@@ -71,14 +75,79 @@ public async Task<AuthenticationResult> ExecuteAsync(
             return await handler.RunAsync(cancellationToken).ConfigureAwait(false);
         }
 
+        private void ValidateAndConfigureMtlsPopForAcquireTokenForClient(
+            AcquireTokenForClientParameters clientParameters, 
+            AcquireTokenCommonParameters commonParameters, 
+            RequestContext requestContext)
+        {
+            if (!clientParameters.UseMtlsPop)
+                return;
+
+            ValidateClaimsOrAssertionsUse();
+            ValidateAuthorityType();
+            ValidateCertificatePresence();
+            ValidateRegionPresence(requestContext);
+
+            commonParameters.MtlsCertificate = _confidentialClientApplication.Certificate;
+            commonParameters.AuthenticationOperation = new MtlsPopAuthenticationOperation(_confidentialClientApplication.Certificate);
+            requestContext.ServiceBundle.Config.IsInstanceDiscoveryEnabled = false;
+            ServiceBundle.Config.ClientCredential = null;
+            requestContext.UseMtlsPop = true;
+        }
+
+        private void ValidateClaimsOrAssertionsUse()
+        {
+            if (_confidentialClientApplication.ServiceBundle.Config.ClaimsOrAssertionsUsed)
+            {
+                ThrowMsalClientException(
+                    MsalError.ClaimsOrAssertionsNotAllowedWithMtlsPop,
+                    MsalErrorMessage.ClaimsAssertionsNotAllowedWithMtlsPop);
+            }
+        }
+
+        private void ValidateAuthorityType()
+        {
+            if (_confidentialClientApplication.AuthorityInfo.AuthorityType != AuthorityType.Aad)
+            {
+                ThrowMsalClientException(
+                    MsalError.InvalidAuthorityType,
+                    MsalErrorMessage.MtlsInvalidAuthorityTypeMessage);
+            }
+        }
+
+        private void ValidateCertificatePresence()
+        {
+            if (_confidentialClientApplication.Certificate == null)
+            {
+                ThrowMsalClientException(
+                    MsalError.MtlsCertificateNotProvided,
+                    MsalErrorMessage.MtlsCertificateNotProvidedMessage);
```

Comment:
Does the certificate need to be passed with x5c? Does that value need to be validated too?